Update package management to go mod

AntonParaskiv opened the pull request:
…ubesim

* Added go mod files
* Updated vendor

AntonParaskiv committed:
* Replaced package path volcano.sh/kubesim -> github.com/volcano-sh/kubesim
* Added go mod files
* Updated vendor

Signed-off-by: Anton Paraskiv <anton@paraskiv.net>

AntonParaskiv committed:
* Replaced package path volcano.sh/kubesim -> github.com/volcano-sh/kubesim
* Added go mod files
* Updated vendor

Signed-off-by: Anton Paraskiv <anton@paraskiv.net>

AntonParaskiv commented:
Hi!
This is solution of issue #7

k82cn commented:
/hold

AntonParaskiv committed:
Merge remote-tracking branch 'origin/issue-7-gomod' into issue-7-gomod

AntonParaskiv committed:
Removed .idea

Signed-off-by: Anton Paraskiv <anton@paraskiv.net>

k82cn commented:
@AntonParaskiv , prefer to keep package to `volcano.sh/kubesim`; similar to `volcano.sh/volcano` :)

k82cn commented on go.mod:
```diff
@@ -0,0 +1,51 @@
+	k8s.io/component-base v0.0.0
+	k8s.io/cri-api v0.0.0
+	k8s.io/klog v1.0.0
+	k8s.io/kubernetes v0.0.0-20200630205207-e38139724f8f
```
let's add version to k8s related repo :)

AntonParaskiv replied:
> let's add version to k8s related repo :)

there is no tag version for commit e38139724f8f0086be9db575d8fe5211c653dc8f, that pointed in README.md
nearest tagged commit is dff82dc0de47299ab66c83c626e08b245ab19037 v1.18.6
should am i use that?

k82cn replied:
`v1.18.6` is ok to me :) please also update other related repo versions :)

k82cn commented on cmd/kubesim/hollow-node.go:
```diff
@@ -45,16 +45,16 @@ import (
-	fakeexec "volcano.sh/kubesim/pkg/mock/util/exec/testing"
-	fakeiptables "volcano.sh/kubesim/pkg/mock/util/iptables/testing"
-	fakesysctl "volcano.sh/kubesim/pkg/mock/util/sysctl/testing"
+	simulatorconfig "github.com/volcano-sh/kubesim/pkg/config"
```
let's keep `volcano.sh` org :)

AntonParaskiv committed:
Updated go mod kubernetes to v1.18.6
Updated vendor
Replaced back module name github.com/volcano-sh -> volcano.sh/kubesim
Updated kubernetes commit revision in README.md

Signed-off-by: Anton Paraskiv <anton@paraskiv.net>

k82cn commented:
/lgtm
/approve

Thanks very much for your contribution!!
